tests/test_core.py

```diff
@@ -1163,21 +1167,36 @@ def test_objcinstance_python_attribute(self):
         self.assertEqual(thing.python_object_1, range(2, 8))
         self.assertEqual(thing.python_object_2, type)
 
+        # Test deleting Python attribute.
+
+        del thing.python_object_1
+
+        with self.assertRaises(AttributeError):
+            thing.python_object_1
+
     def test_objcinstance_python_attribute_keep_alive(self):
         """Python attributes on an ObjCInstance are kept even if the object temporarily has no Python references."""
 
-        Example = ObjCClass("Example")
-        example = Example.alloc().init()
-        Thing = ObjCClass("Thing")
-        thing = Thing.alloc().init()
+        example = NSObject.alloc().init()
+        thing = NSObject.alloc().init()
```

Comment:
I think this changes the behavior of the test quite significantly, along with the change below from `example.setThing(thing)` to `example.thing = thing`. Previously, it tested that:

* If the Objective-C object `thing` has Python attributes
* and it is stored inside an Objective-C property on another object (`example`)
* and `thing` temporarily has *no* references from Python
* once `thing` is retrieved from the Objective-C property, it still has its Python attributes

The new test code is similar, but it stores `thing` in a *Python property* on an Objective-C object, and not in a real Objective-C property. The difference is relevant because when `thing` is stored in a Python property, its Python `ObjCInstance` is kept alive. This test is supposed to test the case where the `ObjCInstance` is temporarily destroyed, but the underlying Objective-C object continues to exist, and is then later used from Python again.

The test as it's written now is probably useful to have as well, but it should be a separate test in addition to the old one, not a replacement.